### PR TITLE
chore: Remove TravisCI references with switch to GitHub actions; update actions versions to remove warnings

### PR DESCRIPTION
**Pull request by bryantbiggs**
- Remove TravisCI references with switch to GitHub actions
- Update actions versions to remove warnings
- Fix mdbook deploy by adding mdbook-linkcheck to workflow

The mdbook stuff could probably be cleaned up and simplified further, but this is a start to get the main CI back green and happy again

### CODE REVIEW

**Comment by bryantbiggs** [comment]
Tested and validated here https://github.com/clowdhaus/lalrpop/actions

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -12,13 +19,19 @@ jobs:
...
-      - name: Install Rust
-        uses: actions-rs/toolchain@v1
+      - uses: actions/checkout@v3
+      - uses: dtolnay/rust-toolchain@master
         with:
           toolchain: ${{ matrix.rust }}
-          override: true
           components: rustfmt
       - name: Run tests
-        run: ./ci.sh
```
Comment (youknowone) [inline .github/workflows/test.yaml:23]: Can we keep those script as separated files? They are helpful when locally testing without CI.

Comment (linkdd) [reply .github/workflows/test.yaml:23]: NB: I often use https://github.com/nektos/act to test my Github Action pipelines.

**Comment by bryantbiggs** [comment]
@youknowone one sec on this - checking the generated mdbook output in my repro to ensure content is generated correctly for github pages. right now it doesn't look like it https://github.com/clowdhaus/lalrpop/tree/gh-pages

**Comment by youknowone** [comment]
#615 is also related to mdbook deployment

**Comment by bryantbiggs** [comment]
ok that fixed it now https://clowdhaus.github.io/lalrpop/

**Comment by bryantbiggs** [comment]
it looks like in #615 they are removing the linkcheck which I think it would be good to keep. Also, this change here is using the mdbook github action which just uses the [latest version to build/publish the mdbook](https://github.com/lalrpop/lalrpop/pull/724/files#diff-28802fbf11c83a2eee09623fb192785e7ca92a3f40602a517c011b947a1822d3R20-R23) which is what is already in use in the `book.yml` workflow

**Review by youknowone — APPROVED** [review]
Thanks a lot!

**Comment by bryantbiggs** [comment]
thank you! Looks like the pages and links are good now 🎉 https://lalrpop.github.io/lalrpop/

**Pull request merged**